pkg/kubelet/cm/internal_container_lifecycle_test.go

```diff
@@ -0,0 +1,121 @@
+package cm
+
+import (
+	"github.com/golang/mock/gomock"
+	"k8s.io/api/core/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	kubefeatures "k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/memorymanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"testing"
+)
+
+func TestPreStartContainer(t *testing.T) {
+	// setup vars
+
+	var tests = []struct {
+		i internalContainerLifecycleImpl
+	}{
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      nil,
+				memoryManager:   nil,
+				topologyManager: nil,
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      cpumanager.NewFakeManager(),
+				memoryManager:   nil,
+				topologyManager: nil,
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      cpumanager.NewFakeManager(),
+				memoryManager:   memorymanager.NewFakeManager(),
+				topologyManager: nil,
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      cpumanager.NewFakeManager(),
+				memoryManager:   nil,
+				topologyManager: topologymanager.NewFakeManager(),
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      nil,
+				memoryManager:   memorymanager.NewFakeManager(),
+				topologyManager: nil,
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      nil,
+				memoryManager:   memorymanager.NewFakeManager(),
+				topologyManager: topologymanager.NewFakeManager(),
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      nil,
+				memoryManager:   nil,
+				topologyManager: topologymanager.NewFakeManager(),
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      cpumanager.NewFakeManager(),
+				memoryManager:   memorymanager.NewFakeManager(),
+				topologyManager: topologymanager.NewFakeManager(),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
```

Comment:
Because we instantiate the mock controller with a `testing.T` we don't need to call Finish - it hooks into the testing.T.

https://pkg.go.dev/github.com/golang/mock/gomock#Controller.Finish

```suggestion
```

Reply:
Yes read in the code comment of go/mock. this is not required for go1.14+ . [ func (ctrl *Controller) Finish() {](https://github.com/golang/mock/blob/aba2ff9a6844d5e3289e8472d3217d5b3090f083/gomock/controller.go#L265)